spylib/utils/hmac.py

```diff
@@ -12,16 +13,12 @@ def calculate_from_message(secret: str, message: str, is_base64: bool = False) -
     return hmac_hash.hexdigest()
 
 
-def calculate_from_components(datetime, path, query_string, body, secret, is_base64: bool = False):
-    if query_string != '':
-        path = path + '?' + query_string
-    message = path + datetime + body
-    return calculate_from_message(secret=secret, message=message, is_base64=is_base64)
+def validate_hmac(secret: str, message: Dict[str, list], is_base64: bool = False):
 
+    hmac_actual = message.pop('hmac')[0]
+    body = '&'.join([f'{arg}={",".join(message[arg])}' for arg in message.keys()])
 
-def validate(secret: str, sent_hmac: str, message: str, is_base64: bool = False):
+    message_hmac = calculate_message_hmac(secret, body, is_base64)
```

Comment:
Named arguments are preferred so it clearer and less error prone
```suggestion
    message_hmac = calculate_message_hmac(secret=secret, message=message, is_base64=is_base64)
```